databricks/koalas/typedef.py

```diff
@@ -133,6 +136,23 @@ def as_python_type(spark_tpe):
     return _py_conversions.get(spark_tpe, None)
 
 
+def infer_pd_series_spark_type(s: pd.Series) -> types.DataType:
+    """Infer Spark DataType from pandas Series dtype.
+
+    :param s: :class:`pandas.Series` to be inferred
+    :return: the inferred Spark data type
+    """
+    dt = s.dtype
+    if dt == np.dtype('object'):
+        if len(s) == 0 or s.isnull().all():
+            raise ValueError("can not infer schema from empty or null dataset")
+        return types.from_arrow_type(pa.Array.from_pandas(s).type)
+    elif is_datetime64_dtype(dt) or is_datetime64tz_dtype(dt):
+        return types.TimestampType()
```

Comment:
@ueshin, sorry if I missed something but why do we need this manual map?

Reply:
Seems like `pa.from_numpy_dtype(dt)` does not work for timestamp types.

Reply:
Ah, given from my rough test, it worked with arrow 0.12. Maybe we can remove it after the upgrade.